src/style-spec/expression/definitions/slice.js

```diff
@@ -0,0 +1,96 @@
+// @flow
+
+import {ValueType, NumberType, toString} from '../types';
+import RuntimeError from '../runtime_error';
+import {typeOf} from '../values';
+
+import type {Expression} from '../expression';
+import type ParsingContext from '../parsing_context';
+import type EvaluationContext from '../evaluation_context';
+import type {Type} from '../types';
+import type {Value} from '../values';
+
+function isSliceableType(type: Type) {
+    return type.kind === 'array' ||
+           type.kind === 'string' ||
+           type.kind === 'value';
+}
+function isAcceptableInputRuntimeValue(input: Array<Value> | string) {
```

Comment:
minot nit: this is the same logic as `isSearchableRuntimeValue`, could you move it to a shared util file.

Sorry if this seems v minor, just trying to be extra strict on bundle size rn.